sdk/dotnet/Pulumi/Resources/CustomResource.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2016-2019, Pulumi Corporation
+
+using Pulumi.Serialization;
+
+namespace Pulumi
+{
+    /// <summary>
+    /// CustomResource is a resource whose create, read, update, and delete(CRUD) operations are
+    /// managed by performing external operations on some physical entity. The engine understands
+    /// how to diff and perform partial updates of them, and these CRUD operations are implemented
+    /// in a dynamically loaded plugin for the defining package.
+    /// </summary>
+    public class CustomResource : Resource
+    {
+        /// <summary>
+        /// Id is the provider-assigned unique ID for this managed resource.  It is set during
+        /// deployments and may be missing (unknown) during planning phases.
+        /// </summary>
+        [Output("id")]
+        public Output<string> Id { get; private set; } = null!;
```

Comment:
this is heavily due to our use of reflection.  We mark it as non-nullable because it won't ever actually be null.  Specifically, the constructor will set all these `[Output(...)]` props to have non-null values.  

However, since that's happens using reflection, the compiler doesn't know about it and complains that no value is ever assigned to the prop. To shut it up, we both assign a value (`null`), and also include `!` to suppress the warning that we're assigning a null to something non-nullable.

It's not pretty.

Reply:
Worth a comment?

Reply:
Sure!